linetools/isgm/abssightline.py

```diff
@@ -0,0 +1,242 @@
+""" Classes for absorption line component
+"""
+from __future__ import print_function, absolute_import, division, unicode_literals
+
+# Python 2 & 3 compatibility
+try:
+    basestring
+except NameError:
+    basestring = str
+
+import pdb
+import numpy as np
+import warnings
+from abc import ABCMeta
+
+from astropy import constants as const
+from astropy import units as u
+from astropy.units import Quantity
+from astropy.coordinates import SkyCoord
+from astropy.table import Table, Column
+
+from linetools.spectralline import AbsLine, SpectralLine
+from linetools import utils as ltu
+from linetools.isgm.abscomponent import AbsComponent
+
+# Global import for speed
+c_kms = const.c.to('km/s').value
+
+# Class for Sightline
+class AbsSightline(object):
+    """ Abstract Class for an absorption sightline
+
+    Attributes
+    ----------
+    name : str
+        Name of the component, e.g. `Si II`
+    coord : SkyCoord
+        Sky coordinate
+    Zion : tuple 
+        Atomic number, ion -- (int,int)
+        e.g. (8,1) for OI
+    zcomp : float
+        Component redshift
+    vlim : Quantity array
+        Velocity limits of the component
+        e.g.  [-300,300]*u.km/u.s
+    A : int
+        Atomic mass -- used to distinguish isotopes
+    Ej : Quantity
+        Energy of lower level (1/cm)
+    comment : str
+        A comment, default is ``
+    """
+    __metaclass__ = ABCMeta
+
+    @classmethod
+    def from_abslines(cls, abslines, stars=None, **kwargs):
+        """Instantiate from a list of AbsLine objects
```

Comment:
This code is duplicated from elsewhere, we may want to move these to utils and call it from there for `AbsSystem` and `AbsComponent` and `AbsSightline`.